Arriba/Arriba/Model/Table.cs

```diff
@@ -591,6 +591,11 @@ public T Query<T>(IQuery<T> query)
         {
             // Route SelectQuery to the Select API. It has a different flow, but users shouldn't have to know not to call Query with SelectQuery instances.
             if (query is SelectQuery) return (T)(object)this.Select((SelectQuery)query);
+            if (query is JoinQuery<SelectResult>)
```

Comment:
This bugs me.  It's already a tiny bit of a compromise that Select doesn't go through the normal Query path.  If you're going to do this, then it seems like it should become genericized to any type which returns a SelectResult.